extensions/coinmarketcap-crypto-crawler/src/index.tsx

```diff
@@ -0,0 +1,183 @@
+  const [cryptoList, setCryptoList] = useState([])
+  const [fuzzyResult, setFuzzyResult] = useState([])
+
+
+  const { push } = useNavigation();
+
+
+  useEffect(() => {
+
+    getListFromFile((err: string, data: string) => {
+      if (err) {
+        console.error(err)
+        return
+      }
+
+      const { cryptoList: cryptoListFromFile } = JSON.parse(data)
+
+      if (!!cryptoListFromFile) {
+        setCryptoList(cryptoListFromFile)
+        return
+      }
+      // fetch crypto list mapping if there's no data exist in the local file
+      // the api has an limit num per request. 
+      fetchAllCrypto({ limit: 10000, start: 1 }).then(({ data: resultData }: { data: ResultData }) => {
+        const { data, status } = resultData
+
+        const cryptoList = data.cryptoCurrencyMap.map(({ slug, name, symbol }: CryptoList) => ({ slug, name, symbol: symbol.toLowerCase() }))
+        writeLIstInToFile({ timestamp: status.timestamp, cryptoList: cryptoList })
+      })
+    })
+  }, [])
+
+  const onSelectCrypto = async (searchText: string) => {
+    const priceInfo = await fetchPrice(searchText) 
+    setIsLoading(false);
+    return priceInfo
+  }
+
+  const onSearchChange = (search:string)=>{
+    setIsLoading(true)
+    setCurrehtSearchText(search);
+
+    const fuzzyResult = fuzzysort.go(search, cryptoList, { key: 'symbol' })
+
+    setFuzzyResult(fuzzyResult)
+  } 
+  
+  return (
+    <List isLoading={isLoading}
+      throttle
+      searchBarPlaceholder="Enter the crypto name"
+      onSearchTextChange={onSearchChange}>
+
+      {
+        fuzzyResult.length === 0 ?
+          <List.Item title="" /> :
+          fuzzyResult.map((result: FuzzySortResult) => {
+            const { name,slug } = result.obj
+
+            return (
+              <List.Item
+                key={name}
+                title={name}
+                icon={Icon.Star}
+                actions={
+                  <ActionPanel>
+                    <ActionPanel.Item title="Pop" onAction={() => {
+                      onSelectCrypto(slug).then(({ currencyPrice = '', priceDiff = '' })=>{
+
+                      push(
+                        <CryptoDetail 
+                            currencyPrice={currencyPrice} 
+                            priceDiff={priceDiff}
+                            name={name}
+                            slug={slug}/>
+                        )  
+                      }) 
+                      
+                    }}/>
+                  </ActionPanel>
+                }
+              />
+            )
+          })
+      }
+
+    </List>
+  );
+}
+
+
+
+
+function CryptoDetail({ currencyPrice, priceDiff,name,slug }: CryptoInfo) {
+  const { pop } = useNavigation();
+  return (
+    <Detail
+      navigationTitle={name}
+      markdown={`## ${currencyPrice} \n ### ${priceDiff}`}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Item title="Pop" onAction={pop} />
```

Comment:
I think the "Pop" action is redundant here. There are back button and escape button makes the thing.
```suggestion
```